src/Microsoft.OData.Core/UriParser/Aggregation/ApplyBinder.cs

```diff
@@ -116,6 +129,7 @@ private IEdmTypeReference CreateAggregateExpressionTypeReference(SingleValueNode
                                     expressionPrimitiveKind));
                     }
 
+                case AggregationMethod.VirtualPropertyCount:

```

Comment:
The change looks good.  However, it looks like both CountDistinct and VirtualPropertyCount should return Decimal not Int64.  I don't think you should change CountDistinct as part of your change, could you simply add a TODO comment that these should be Decimal instead of Int64?  Something like:

    // TODO: CountDistinct and $Count should return type Edm.Decimal with Scale="0" and sufficient Precision.
    return EdmCoreModel.Instance.GetPrimitive(EdmPrimitiveTypeKind.Int64, false); 

3.1.3.5 Standard Aggregation Method countdistinct
The result property MUST have type Edm.Decimal with Scale="0" and sufficient Precision.

3.1.5 Virtual Property $count
The result property will have type Edm.Decimal with Scale="0" and sufficient Precision.